File: ReactRedux/src/models/ModelMetadata.ts
import { PropMetadata, initPropMetadata } from "./PropMetadata";
import { ProjectMetadata, initProjectMetadata } from "./ProjectMetadata";
export interface ModelMetadata {
  idModelMetadata:number;
  name:string;
  initData?:string;
  nameSpace?:string;
  caption?:string;
  props:PropMetadata[];
  idProjectMetadata:number;
  projectMetadata?:ProjectMetadata;
}

export const initModelMetadata = {
  idModelMetadata: 0,
  name: '',
  props: [],
  idProjectMetadata: 0,

}

File: DataAccessLayer/Models/ModelMetadata.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DataAccessLayer.Dto
{
    public class ModelMetadata
    {

        public ModelMetadata()
        {
          Props = new HashSet<PropMetadata>();

        }

        [Key]
        public int IdModelMetadata { get; set; }
        public string Name { get; set; }
        public string? InitData { get; set; }
        public string? NameSpace { get; set; }
        public string? Caption { get; set; }

        
        public virtual ICollection<PropMetadata>? Props { get; set; }
        public int? IdProjectMetadata { get; set; }

        
        [ForeignKey("IdProjectMetadata")]
        public virtual ProjectMetadata? ProjectMetadata { get; set; }
    }
}
